app/helpers/application_helper.rb

```diff
@@ -675,6 +675,23 @@ def sort_link_direction(column)
     params[:sort].eql?(column) && params[:direction].eql?("asc") ? "desc" : "asc"
   end
 
+  def search_path(opts = {})
+    o = opts.dup
+    o.delete("controller")
+    o.delete("action")
+
+    case [CustomLandingPage::LandingPageStore.enabled?(@current_community&.id), params[:locale].present?]
```

Comment:
There is a bit similar logic in `application_controller#set_homepage_path` method. That method takes also `@current_user` into account. I.e. it doesn't add the locale, even if it's in the parameters, if `@current_user` exists, because the locale is already in the session. Does that make sense? Should we add it here also?